src/lexer/attribute.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (C) 2018 Glayzzle (BSD3 License)
+ * @authors https://github.com/glayzzle/php-parser/graphs/contributors
+ * @url http://glayzzle.com
+ */
+"use strict";
+
+module.exports = {
+  matchST_ATTRIBUTE: function () {
+    let listDepth = 0;
+    let ch = this.input();
+    if (this.is_WHITESPACE()) {
+      do {
+        ch = this.input();
+      } while (this.is_WHITESPACE());
+      this.unput(1);
+      return null;
+    }
+    // process.stderr.write("Token:" + JSON.stringify(ch) + "\n");
```

Comment:
Not needed

Reply:
For reasons that I dont understand, if I dont unput and return null, then all the Attribute names include leading spaces & sometimes newlines, eg:
```diff
              Attribute {
                "args": Array [],
                "kind": "attribute",
-               "name": "Two",
+               "name": "
+       Two",
              },
```